### DISCUSSION TITLE
Paid support options

### DISCUSSION THREAD

**Discussion by mliberty1** [pyqtgraph/pyqtgraph#1883] [Ideas]
This "issue" collects options and user interest regarding paid work on pyqtgraph.  Why?  pyqtgraph is awesome.  Some of us use pyqtgraph for commercial work (see [Joulescope](https://www.joulescope.com/), [GitHub](https://github.com/jetperch/pyjoulescope_ui/tree/master/joulescope_ui/widgets/waveform)), and we are in the fortunate position to be able to pay for features and support.  All things being equal, I would rather give money to the pyqtgraph project maintainers than some random Python consultant.

What are some of the options?

* **Consulting**: Pay project maintainers directly to perform work.  The maintainers can leverage their expertise to deliver higher quality in less time compared to other consultants.  It also allows maintainers to develop code using pyqtgraph, not just within pyqtgraph.  From the user perspective, this option is easiest, but it can be "transactional" without building up a long-term
relationship.  Another potential downside is that pyqtgraph becomes a job, not a hobby or labor of love, which can degrade long-term desire for some maintainers.
* **Sponsorship**: Pay a monthly or annual fee to pyqtgraph.  Sponsorship gives the most long-term stability and control to the project maintainers of these options.  However, sponsors must still get value.  Options for delivering value include:
   * Listing as a sponsor on the project website.  Good for attracting small $ sponsorships.
   * Prioritized support to bugs and feature requests - can create conflict.
   * N "chunks" of work for each contribution level - can create a "consulting-style" relationship.
   I have a friend that is on the core [Nerves](https://www.nerves-project.org) team, which uses [OpenCollective](https://opencollective.com/nerves-project).  I'll ask about his experience. I can connect you, if you would like. 
* **[Bounty program](https://en.wikipedia.org/wiki/Open-source_bounty)**: A user offers a cash reward for a feature.  Community members can decide to accept or not.  The downside is that bounties have no long-term goal and may create features (and long-term support) that do not benefit the community.  I have no experience with collecting & managing a bounty program.  After doing more [research](https://duo.com/decipher/open-source-software-needs-funding-not-bug-bounty-programs), it seems that bounty programs cause more problems than they solve.  

I think that sponsorship and consulting are not exclusive.  I am definitely open to supporting both.

What do you think?

**Comment by j9ac9k** [comment]
Hi @mliberty1 

First off, thanks for sharing joulescope!  Over the last ~3-4 years, us maintainers have become _very_ detached from our users, and collectively we had little idea what pyqtgraph was used for outside of https://github.com/acq4 and https://github.com/biolab/orange3  

> All things being equal, I would rather give money to the pyqtgraph project maintainers than some random Python consultant.

I appreciate this sentiment; I can't speak for all of the maintainers, but many of us already have full time jobs; and I'd be reluctant to take much in the way of funding as while I am a maintainer, some of the best code changes have been made by contributors, _not_ maintainers (some have since become maintainers, and still do awesome work).   This for me sort of rules out sponsorship, consulting will vary on a case-by-case basis for maintainers.  The possible exception here is there is sufficient demand for a kind of CI service we cannot get on a free tier (say benchmark testing involving tasks that run on GPUs for example).  

Of the three options, I _really_ like the bounty-program the best.  Only problem here is that I've never run one, have no idea what's involved, or what I should be aware of.  If anyone here has experience running one and would like to chime in, I'd love to hear about it.

**Comment by lyl1836** [comment]
Support and willing to pay consulting fee！
